target/linux/lantiq/dts/RS353.dts

```diff
@@ -0,0 +1,318 @@
+/dts-v1/;
+
+/include/ "vr9.dtsi"
+
+/ {
+	model = "RS353 - Bintec Elmeg RS353";
+
+	chosen {
+		bootargs = "console=ttyLTQ0,115200 init=/etc/preinit";
+		leds {
+                        running = &status;
+                        boot = &status;
+			wifi = &wlan;
+			usb2 = &usb; /* Front port is mapped to device 2 */
+			dsl = &dsl;
+                };
+
+	};
+
+	memory@0 {
+		reg = <0x0 0x8000000>;
+	};
+
+	sram@1F000000 {
+		cgu0: cgu@103000 {
+                        compatible = "lantiq,cgu-xway";
+                        reg = <0x103000 0x1000>;
+                        #clock-cells = <1>;
+			lantiq,phy-clk-src = <0x2>;
+		};
+	};
+
+	fpi@10000000 {
+		localbus@0 {
+			nor-boot@0 {
+				compatible = "lantiq,nor";
+				bank-width = <2>;
+				reg = <0 0x0 0x2000000>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+
+				partitions {
+					compatible = "fixed-partitions";
+					#address-cells = <1>;
+					#size-cells = <1>;
+
+					boardconfig: partition@0 {
```

Comment:
OK I have a good news :)

I got it working by disabling the endianness workaround by commenting the following code:

```
if (map->map_priv_1 == LTQ_NOR_PROBING)
             adr ^= 2;
```

Now, I have: 

```
OpenWrt kernel loader for MIPS based SoC
Copyright (C) 2011 Gabor Juhos <juhosg@openwrt.org>
Decompressing kernel... done!
Starting kernel at 80002000...

[    0.000000] Linux version 4.4.14 (qa@serveurQA) (gcc version 5.3.0 (OpenWrt GCC 5.3.0 50020) ) #30 SMP Wed Jan 11 08:33:14 UTC 2017
[    0.000000] SoC: xRX200 rev 1.2
[    0.000000] bootconsole [early0] enabled
[    0.000000] CPU0 revision is: 00019556 (MIPS 34Kc)
[    0.000000] MIPS: machine is RS353 - Bintec Elmeg RS353
[    0.000000] Determined physical RAM map:
[    0.000000]  memory: 08000000 @ 00000000 (usable)
[    0.000000] Initrd not found or empty - disabling initrd
[    0.000000] Zone ranges:
[    0.000000]   Normal   [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] Detected 1 available secondary CPU(s)
[    0.000000] Primary instruction cache 32kB, VIPT, 4-way, linesize 32 bytes.
[    0.000000] Primary data cache 32kB, 4-way, VIPT, cache aliases, linesize 32 bytes
[    0.000000] PERCPU: Embedded 9 pages/cpu @8110c000 s8128 r8192 d20544 u36864
[    0.000000] Built 1 zonelists in Zone order, mobility grouping on.  Total pages: 32512
[    0.000000] Kernel command line: console=ttyLTQ0,115200 init=/etc/preinit
[    0.000000] PID hash table entries: 512 (order: -1, 2048 bytes)
[    0.000000] Dentry cache hash table entries: 16384 (order: 4, 65536 bytes)
[    0.000000] Inode-cache hash table entries: 8192 (order: 3, 32768 bytes)
[    0.000000] Writing ErrCtl register=00048501
[    0.000000] Readback ErrCtl register=00048501
[    0.000000] Memory: 122972K/131072K available (4019K kernel code, 168K rwdata, 1172K rodata, 1216K init, 222K bss, 8100K reserved, 0K cma-reserved)
[    0.000000] SLUB: HWalign=32, Order=0-3, MinObjects=0, CPUs=2, Nodes=1
[    0.000000] Hierarchical RCU implementation.
[    0.000000] NR_IRQS:256
[    0.000000] Setting up vectored interrupts
[    0.000000] Setting up IPI vectored interrupts
[    0.000000] CPU Clock: 500MHz
[    0.000000] clocksource: MIPS: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 7645041786 ns
[    0.000011] sched_clock: 32 bits at 250MHz, resolution 4ns, wraps every 8589934590ns
[    0.007839] Calibrating delay loop... 332.54 BogoMIPS (lpj=665088)
[    0.042407] pid_max: default: 32768 minimum: 301
[    0.047254] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.053799] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.064799] Primary instruction cache 32kB, VIPT, 4-way, linesize 32 bytes.
[    0.064816] Primary data cache 32kB, 4-way, VIPT, cache aliases, linesize 32 bytes
[    0.065067] CPU1 revision is: 00019556 (MIPS 34Kc)
[    0.124689] Synchronize counters for CPU 1: done.
[    0.129575] Brought up 2 CPUs
[    0.137109] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 7645041785100000 ns
[    0.146960] pinctrl core: initialized pinctrl subsystem
[    0.152958] NET: Registered protocol family 16
[    0.163541] pinctrl-xway 1e100b10.pinmux: Init done
[    0.169594] dma-xway 1e104100.dma: Init done - hw rev: 7, ports: 7, channels: 28
[    0.282805] dcdc-xrx200 1f106a00.dcdc: Core Voltage : 1016 mV
[    0.445496] usbcore: registered new interface driver usbfs
[    0.451126] usbcore: registered new interface driver hub
[    0.456639] usbcore: registered new device driver usb
[    0.462463] PCI host bridge to bus 0000:00
[    0.466575] pci_bus 0000:00: root bus resource [mem 0x1c000000-0x1cffffff]
[    0.473473] pci_bus 0000:00: root bus resource [io  0x1d800000-0x1d8fffff]
[    0.480397] pci_bus 0000:00: root bus resource [??? 0x00000000 flags 0x0]
[    0.487210] pci_bus 0000:00: No busn resource found for root bus, will use [bus 00-ff]
[    0.495254] ifx_pcie_rc_class_early_fixup: fixed pcie host bridge to pci-pci bridge
[    0.512410] pci 0000:00:00.0: BAR 8: assigned [mem 0x1c000000-0x1c0fffff]
[    0.519171] pci 0000:00:00.0: BAR 9: assigned [mem 0x1c100000-0x1c1fffff pref]
[    0.526431] pci 0000:01:00.0: BAR 0: assigned [mem 0x1c000000-0x1c01ffff 64bit]
[    0.533793] pci 0000:01:00.0: BAR 6: assigned [mem 0x1c100000-0x1c10ffff pref]
[    0.541015] pci 0000:00:00.0: PCI bridge to [bus 01]
[    0.546032] pci 0000:00:00.0:   bridge window [mem 0x1c000000-0x1c0fffff]
[    0.552856] pci 0000:00:00.0:   bridge window [mem 0x1c100000-0x1c1fffff pref]
[    0.560133] ifx_pcie_bios_map_irq port 0 dev 0000:00:00.0 slot 0 pin 1 
[    0.566746] ifx_pcie_bios_map_irq dev 0000:00:00.0 irq 144 assigned
[    0.573095] ifx_pcie_bios_map_irq port 0 dev 0000:01:00.0 slot 0 pin 1 
[    0.579706] ifx_pcie_bios_map_irq dev 0000:01:00.0 irq 144 assigned
[    0.587755] clocksource: Switched to clocksource MIPS
[    0.595497] NET: Registered protocol family 2
[    0.600820] TCP established hash table entries: 1024 (order: 0, 4096 bytes)
[    0.607790] TCP bind hash table entries: 1024 (order: 1, 8192 bytes)
[    0.614147] TCP: Hash tables configured (established 1024 bind 1024)
[    0.620638] UDP hash table entries: 256 (order: 1, 8192 bytes)
[    0.626464] UDP-Lite hash table entries: 256 (order: 1, 8192 bytes)
[    0.633087] NET: Registered protocol family 1
[    0.637982] gptu: totally 6 16-bit timers/counters
[    0.642882] gptu: misc_register on minor 63
[    0.647096] gptu: succeeded to request irq 126
[    0.651579] gptu: succeeded to request irq 127
[    0.656090] gptu: succeeded to request irq 128
[    0.660566] gptu: succeeded to request irq 129
[    0.665057] gptu: succeeded to request irq 130
[    0.669550] gptu: succeeded to request irq 131
[    0.675611] phy-xrx200 gphy-xrx200: requesting lantiq/vr9_phy11g_a2x.bin
[    0.683036] phy-xrx200 gphy-xrx200: booting GPHY0 firmware at 7A20000
[    0.689455] phy-xrx200 gphy-xrx200: booting GPHY1 firmware at 7A20000
[    0.774468] futex hash table entries: 512 (order: 2, 16384 bytes)
[    0.799795] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.805582] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    0.818955] io scheduler noop registered
[    0.823048] io scheduler deadline registered (default)
[    0.829302] 1e100c00.serial: ttyLTQ0 at MMIO 0x1e100c00 (irq = 112, base_baud = 0) is a lantiq,asc
[    0.838297] console [ttyLTQ0] enabled
[    0.838297] console [ttyLTQ0] enabled
[    0.845633] bootconsole [early0] disabled
[    0.845633] bootconsole [early0] disabled
[    0.855511] lantiq nor flash device: 02000000 at 10000000
[    0.860965] ltq_nor: Found 1 x16 devices at 0x0 in 16-bit bank. Manufacturer ID 0x0000c2 Chip ID 0x00227e
[    0.870493] Amd/Fujitsu Extended Query Table at 0x0040
[    0.875641]   Amd/Fujitsu Extended Query version 1.3.
[    0.880685] number of CFI chips: 1
[    0.884138] 8 ofpart partitions found on MTD device ltq_nor
[    0.889660] Creating 8 MTD partitions on "ltq_nor":
[    0.894541] 0x000000000000-0x000000020000 : "logic_config"
[    0.902163] 0x000000020000-0x000000040000 : "logic_addr"
[    0.909540] 0x000000040000-0x000000060000 : "bootmonitor"
[    0.917470] 0x000000080000-0x0000000c0000 : "flash_config"
[    0.924879] 0x0000000c0000-0x000000100000 : "uboot"
[    0.932028] 0x000000100000-0x000000120000 : "uboot-env"
[    0.939326] 0x000000120000-0x000001fe0000 : "firmware"
[    1.037950] 0x000001fe0000-0x000002000000 : "GPHY_CPR"
[    1.152864] libphy: lantiq,xrx200-mdio: probed
[    1.224801] eth0: attached PHY [Lantiq XWAY VR9 GPHY 11G v1.4] (phy_addr=0:13, irq=-1)
[    1.296798] eth0: attached PHY [Lantiq XWAY VR9 GPHY 11G v1.4] (phy_addr=0:11, irq=-1)
[    1.368831] eth0: attached PHY [Lantiq XWAY PEF7071] (phy_addr=0:00, irq=-1)
[    1.440814] eth0: attached PHY [Lantiq XWAY PEF7071] (phy_addr=0:01, irq=-1)
[    1.512813] eth0: attached PHY [Lantiq XWAY PEF7071] (phy_addr=0:02, irq=-1)
[    1.521864] wdt 1f8803f0.watchdog: Init done
[    1.530800] NET: Registered protocol family 10
[    1.536963] NET: Registered protocol family 17
[    1.541505] bridge: automatic filtering via arp/ip/ip6tables has been deprecated. Update your scripts to load br_netfilter if you need this.
[    1.554065] 8021q: 802.1Q VLAN Support v1.8
[    1.564266] VFS: Cannot open root device "(null)" or unknown-block(0,0): error -6
[    1.570363] Please append a correct "root=" boot option; here are the available partitions:
[    1.578720] 1f00             128 mtdblock0  (driver?)
[    1.583768] 1f01             128 mtdblock1  (driver?)
[    1.588759] 1f02             128 mtdblock2  (driver?)
[    1.593811] 1f03             256 mtdblock3  (driver?)
[    1.598859] 1f04             256 mtdblock4  (driver?)
[    1.603962] 1f05             128 mtdblock5  (driver?)
[    1.608956] 1f06           31488 mtdblock6  (driver?)
[    1.614005] 1f07             128 mtdblock7  (driver?)
[    1.619045] Kernel panic - not syncing: VFS: Unable to mount root fs on unknown-block(0,0)
[    1.629805] Rebooting in 10 seconds..
```

The flash content is now read in the same way as the Bootmonitor does. This explains that the filesystem isn't detected  well. This means the entire OpenWrt system can now be flahshed by the Original OS, and I don't need to hack anymore to read the MAC addr in the flash memory.